src/lang/base/ScillaParser.mly

```diff
@@ -23,21 +23,43 @@
 
   open ParsedSyntax
 
-  let to_type d = match d with
-    | x when PrimTypes.is_prim_type (PrimType x) -> PrimType x
-    | _ -> ADT (d, [])
-         
-  let to_map_key_type d = match d with
-    | x
-         when PrimTypes.is_prim_type (PrimType x) &&
-                not (PrimType x = PrimTypes.msg_typ || PrimType x = PrimTypes.event_typ)
-      -> PrimType x
-    | _ -> raise (SyntaxError ("Invalid map key type " ^ d))
+  let to_prim_type_exn d = match d with
+    | "Int32" -> Int_typ Bits32
+    | "Int64" -> Int_typ Bits64
+    | "Int128" -> Int_typ Bits128
+    | "Int256" -> Int_typ Bits256
+    | "Uint32" -> Uint_typ Bits32
+    | "Uint64" -> Uint_typ Bits64
+    | "Uint128" -> Uint_typ Bits128
+    | "Uint256" -> Uint_typ Bits256
+    | "String" -> String_typ
+    | "BNum" -> Bnum_typ
+    | "Message" -> Msg_typ
+    | "Event" -> Event_typ
+    | "ByStr" -> Bystr_typ
+    | _ -> let re = Str.regexp "ByStr\\([0-9]+\\)$" in
```

Comment:
This check for `ByStrX` is probably better of as a separate function (maybe somewhere in `PrimTypes.ml`)

Reply:
On second thoughts, it may be fine the way it is. I leave it to you and other reviewers.